providers/fcontext.rb

```diff
@@ -0,0 +1,53 @@
+# Support whyrun
+def whyrun_supported?
+  true
+end
+
+use_inline_resources
+
+# Create if doesn't exist, do not touch if fcontext is already registered
+action :add do
+  if use_selinux then
+    escaped_file_spec = Regexp.escape(new_resource.file_spec)
+    execute "selinux-fcontext-#{new_resource.secontext}-add" do
+      command lazy { "/usr/sbin/semanage fcontext -a -t #{new_resource.secontext} '#{new_resource.file_spec}' && #{restorecon(new_resource.file_spec)}" }
+      not_if "/usr/sbin/semanage fcontext -l | grep -P '#{escaped_file_spec}.*#{new_resource.secontext}'>/dev/null"
+    end
+  end
+end
+
+# Delete if exists
+action :delete do
+  if use_selinux then
+    escaped_file_spec = Regexp.escape(new_resource.file_spec)
+    execute "selinux-fcontext-#{new_resource.secontext}-delete" do
+      command lazy { "/usr/sbin/semanage fcontext -d '#{new_resource.file_spec}' && #{restorecon(new_resource.file_spec)}" }
+      only_if "/usr/sbin/semanage fcontext -l | grep -P '#{escaped_file_spec}.*#{new_resource.secontext}'>/dev/null"
```

Comment:
Regex should include spaces (`#{a}\s.*\s#{b}` or something) so it'll be less icky